serializer/serializer.go

```diff
@@ -284,16 +297,12 @@ func (s *Serializer) WriteObject(seri Serializable, deSeriMode DeSerializationMo
 
 // WriteSliceOfObjects writes Serializables into the Serializer.
 // For every written Serializable, the given WrittenObjectConsumer is called if it isn't nil.
-func (s *Serializer) WriteSliceOfObjects(seris Serializables, deSeriMode DeSerializationMode, woc WrittenObjectConsumer, errProducer ErrProducer) *Serializer {
+func (s *Serializer) WriteSliceOfObjects(seris Serializables, deSeriMode DeSerializationMode, lenType SeriSliceLengthType, woc WrittenObjectConsumer, errProducer ErrProducer) *Serializer {
 	if s.err != nil {
 		return s
 	}
 
-	if err := binary.Write(&s.buf, binary.LittleEndian, uint16(len(seris))); err != nil {
-		s.err = errProducer(err)
-		return s
-	}
-
+	_ = s.writeSliceLength(len(seris), lenType, errProducer)
```

Comment:
Check here whether an error was produced by checking `if s.err != nil ...` as there is no point in doing further work if the slice length couldn't be written.